### PR TITLE
Adding in recent change to Vyper syntax

### PR DESCRIPTION
**Pull request by tpmccallum**
Issue https://github.com/ethereum/vyper/issues/695 has been resolved with the Vyper log event syntax becoming more aligned with the traditional Solidity syntax. 

Also explained the difference between state and chain data (which needs some more info soon) and preliminary explanation of how to write to, and read from, logs.

### CODE REVIEW

File: contrib/vyper.asciidoc [hunk 1]
```diff
@@ -47,6 +47,13 @@ The human readable ABI code (in JSON format) can be obtained by then running the
 
 vyper -f json ~/hello_world.v.py
 
+=== Reading and writing data
+Ethereum smart contracts can write data to two places. Ethereum's global state trie or Ethereum's chain data. While it is costly to store, read and modify data, these storage operations are a necessary component of most smart contracts. 
 
+==== Global state
+The state variables in a given Ethereum smart contract are stored in Ethereum's global state trie. A given smart contract can only store, read and modify data specifically in relation to that contracts address i.e. smart contracts can not read or write to other smart contracts. TODO continue with more information about this in relation to Vyper
```
Comment (come-maiz) [inline contrib/vyper.asciidoc]: You are missing a ', and I would put the ie in parentheses.
contract's address (i.e. smart contracts ... )

File: contrib/vyper.asciidoc [hunk 2]
```diff
@@ -47,6 +47,13 @@ The human readable ABI code (in JSON format) can be obtained by then running the
...
 
+==== Log
+As previously mentioned, a smart contract can also write to Ethereum's chain data through log events. Whilst Vyper initially employed the __log__ syntax for declaring these events, an update has been made which brings Vypers event declaration more in line with Solidity's original syntax. For example, Vyper's declaration of an event called MyLog was originally MyLog: __log__({arg1: indexed(bytes <= 3)}) Vyper's syntax has now become MyLog: event({arg1: indexed(bytes <= 3)}). It is important to note that the execution of the log event in Vyper was and still is as follows log.MyLog("123").
```
Comment (come-maiz) [inline contrib/vyper.asciidoc]: "Whilst" is not very common, at least for non-native speakers like me. "While" makes it more readable, I think.

Comment (come-maiz) [reply contrib/vyper.asciidoc]: Is it useful to explain how it previously was done? I think it would be enough to explain how it is done now. Less noise.

**Comment by tpmccallum** [comment]
Hi @elopio 
I hope that you are happy with my recent changes. 
Please let me know if you have any more ideas.
I will be making more PRs as I work through all of the TODOs which I put in the original file a while back.
Many thanks
Tim

File: contrib/vyper.asciidoc [hunk 3]
```diff
@@ -47,6 +47,13 @@ The human readable ABI code (in JSON format) can be obtained by then running the
...
 
+==== Global state
+The state variables in a given Ethereum smart contract are stored in Ethereum's global state trie. A given smart contract can only store, read and modify data specifically in relation to that contract's address (i.e. smart contracts can not read or write to other smart contracts). TODO continue with more information about this in relation to Vyper
 
+==== Log
+As previously mentioned, a smart contract can also write to Ethereum's chain data through log events. While Vyper initially employed the __log__ syntax for declaring these events, an update has been made which brings Vypers event declaration more in line with Solidity's original syntax. For example, Vyper's declaration of an event called MyLog was originally MyLog: __log__({arg1: indexed(bytes <= 3)}) Vyper's syntax has now become MyLog: event({arg1: indexed(bytes <= 3)}). It is important to note that the execution of the log event in Vyper was and still is as follows log.MyLog("123").
```
Comment (come-maiz) [inline contrib/vyper.asciidoc]: Vyper's event declaration

**Comment by come-maiz** [comment]
one more typo, and the rest looks good to me @tpmccallum

**Comment by tpmccallum** [comment]
Thank you @elopio 
All done, good to go.
I will return with more!

**Pull request merged**